helios-services/src/main/java/com/spotify/helios/agent/TaskMonitor.java

```diff
@@ -242,6 +243,20 @@ private void updateState(final TaskStatus.State state) {
   }
 
   /**
+   * Propagate the container error by setting it and committing the update.
+   */
+  private void updateContainerError(final String containerError) {
+    statusUpdater.setContainerError(containerError);
+    // Commit and push a new status
+    try {
+      statusUpdater.update();
+    } catch (InterruptedException e) {
+      // TODO: propagate interrupt instead?
+      Thread.currentThread().interrupt();
```

Comment:
It seems like the updateState() method is a convenience method so the caller doesn't need to remember to call `StatusUpdater.update()` and catch an `InterruptedException`. Maybe instead of

```
    updateState(FAILED);
    updateContainerError(containerError);
 ```
you could do

```
    statusUpdater.setState(FAILED);
    statusUpdater.setContainerError(containerError);
    statusUpdater.update();
```